### ISSUE TITLE
dropout backend function wrong ?

### ISSUE THREAD

**Issue by SleepProgger** [plaidml/plaidml#283]
The tile code for the backend function `dropout` looks like:
```
function (I, X, L) -> (O) {
        R = 1.0 - L;
        M = 1.0 / R;
        O = (prng_value(I) < R ? X * M : 0.0);
    }
```

and as keras `Dropout` layer simply returns the result of the dropout function this leads to not only to values dropped (like expected) but also to a multiplication with `(1 / (1-Level))` for not dropped values.
If my understanding is correct that is the wrong behavior.
Simple fix is replace `X * M` with a simple `X`

**Demo:**
```
    in_ = v = keras.layers.Input((5, 5, 1), dtype="float32")
    v = keras.layers.Dropout(0.5)(v)
    func = K.function([in_, K.learning_phase()],[v])
    print(func((np.ones((1, 5, 5, 1), dtype="float32"), 1)))
```

**Result**
```
[array([[[[2.],
         [0.],
         [0.],
         [0.],
         [2.]],

        [[2.],
         [0.],
         [0.],
         [0.],
         [0.]],

        [[2.],
         [0.],
         [0.],
         [0.],
         [2.]],

        [[2.],
         [2.],
         [2.],
         [0.],
         [2.]],

        [[0.],
         [0.],
         [2.],
         [0.],
         [0.]]]], dtype=float32)]
```

**Comment by SleepProgger** [comment]
Ok, so from what i understand, after reading more paper and stackoverflow, keras uses "inverse dropout" where this behavior is indeed correct. 
Once again, sorry for the noise.
